compiler/z/codegen/S390GenerateInstructions.cpp

```diff
@@ -1094,81 +1109,156 @@ TR::Instruction *
 generateRSInstruction(TR::CodeGenerator * cg, TR::InstOpCode::Mnemonic op, TR::Node * n, TR::Register * treg, TR::MemoryReference * mf,
                       TR::Instruction * preced)
    {
-   // non-Trex cannot upgrade to Y form so take this opportunity to fold in a large offset
-   if (mf) preced = mf->separateIndexRegister(n, cg, false, preced);
+   // RS and RSY instructions do not have an index register
+   preced = mf->separateIndexRegister(n, cg, false, preced);
 
-   TR::Instruction *instr;
-   if (preced)
-      instr = new (INSN_HEAP) TR::S390RSInstruction(op, n, treg, mf, preced, cg);
-   else
-      instr = new (INSN_HEAP) TR::S390RSInstruction(op, n, treg, mf, cg);
+   TR::Instruction* result = NULL;
 
-   return instr;
+   int32_t displacement = mf->getOffset();
```

Comment:
This pattern repeats across the various generateRS instructions (or at least it appears to be very similar).  Can it be pulled into a separate function to save some code space?

Reply:
Fixed in 5c826f3